python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -270,6 +270,48 @@ def compute_child_subset(
                 value=_ValidatedEntitySubsetValue(child_partitions_subset),
             )
 
+    def compute_map_subset(
+        self, to_key: T_EntityKey, subset: EntitySubset[T_EntityKey]
```

Comment:
`from_subset` should be of type `EntitySubset[U_EntityKey]`, as we want to support mapping between different types -- as written, this forces to_key / from_subset to either both be AssetKey or both be AssetCheckKey.

I think this change will let you get rid of the casts you have in other parts of the code.